src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/CngAsnFormatter.cs

```diff
@@ -17,9 +18,38 @@ protected override string FormatNative(Oid oid, byte[] rawData, bool multiLine)
             // to use hex formatting by providing an empty OID string.
             string oidValue = string.Empty;
             if (oid != null && oid.Value != null)
+            {
                 oidValue = oid.Value;
+            }
 
-            return Cng.CryptFormatObject(oidValue, rawData, multiLine);
+            int dwFormatStrType = multiLine ? Interop.Crypt32.CRYPT_FORMAT_STR_MULTI_LINE : Interop.Crypt32.CRYPT_FORMAT_STR_NONE;
+            int cbFormat = 0;
+            const int X509_ASN_ENCODING = 0x00000001;
+            unsafe
+            {
+                IntPtr oidValuePtr = Marshal.StringToHGlobalAnsi(oidValue);
```

Comment:
Why do this instead of LPStr marshal the string?

Reply:
I wanted to share the single P/Invoke signature (that closely matches the Win32 definition), and the Win32 definition is crazy: based on the value of the pointer, it's either treated as an actual pointer or it's treated as a known constant.  And the other call site for this passes in 7.  So to use a marshaling directive I would need to duplicate the DllImports.